cve_bin_tool/cve_scanner.py

```diff
@@ -180,18 +181,16 @@ def get_cves(self, vendor: str, product: str, version: str, remarks: Remarks):
                 self.products_without_cve += 1
             self.all_cves.add(cve_data)
 
-    @staticmethod
-    def openssl_convert(version: str) -> str:
+    def openssl_convert(self, version: str) -> str:
         """ pkg_resources follows pep-440 which doesn't expect openssl style 1.1.0g version numbering
         So to fake it, if the last character is a letter, replace it with .number before comparing """
-        if len(version) < 1:
+        if not version:  # if version is empty return it.
             return version
 
-        lastchar = version[len(version) - 1]
-        letters = dict(zip(ascii_lowercase, range(26)))
+        lastchar = version[-1]
 
-        if lastchar in letters:
-            version = f"{version[0: len(version) - 1]}.{letters[lastchar]}"
+        if lastchar in self.ALPHA_TO_NUM:
+            version = f"{version[:-1]}.{self.ALPHA_TO_NUM[lastchar]}"
```

Comment:
If we keep ALPHA_TO_NUM in function whenever that function get called new dictionary will get allocated in function stack and it will be destroyed after function returns. Since we are going to call this function several times with CVEScanner context it will be more efficient to keep it as a class attribute.

Also, I noticed `version[0: len(version) - 1]` wasn't idiomatic so I replaced it with `version[:-1]` which is more pythonic.